<!-- img_proc.launch
  This file includes the img_proc_full node and visualization tools.
  Author: Francisco J. Perez Grau (fjperezgrau@gmail.com)
-->

<launch>

  <node name="img_proc" type="img_proc_full" pkg="unia_perception" output="screen">
    <param name = "image_topic" value = "/camera/rgb/image_raw" />
    <param name = "caminfo_topic" value = "/camera/rgb/camera_info" />
    <param name = "h_min" value = "18" />		# Min value in channel H for inRange
    <param name = "h_max" value = "35" />		# Max value in channel H for inRange
    <param name = "s_min" value = "90" />		# Min value in channel S for inRange
    <param name = "s_max" value = "255" />		# Max value in channel S for inRange
    <param name = "v_min" value = "70" />		# Min value in channel V for inRange
    <param name = "v_max" value = "255" />		# Max value in channel V for inRange
    <param name = "morph_size" value = "10" />		# Size for the StructuringElement in erode and dilate
    <param name = "target_radius" value = "0.085" />	# Radius of the target ball
  </node>

  <node name="source" pkg="image_view" type="image_view" args="image:=/camera/rgb/image_raw" />
  <node name="hsv" pkg="image_view" type="image_view" args="image:=/image_processor/hsv_image" />
  <node name="thresholded" pkg="image_view" type="image_view" args="image:=/image_processor/thres_image" />
  <node name="morph_op" pkg="image_view" type="image_view" args="image:=/image_processor/morph_image" />
  <node name="contours" pkg="image_view" type="image_view" args="image:=/image_processor/contour_image" />
  <node name="processed" pkg="image_view" type="image_view" args="image:=/image_processor/det_image" />

  <node name="rviz" type="rviz" pkg="rviz" args="--d $(find unia_perception)/launch/img_proc.rviz" output="screen" />

</launch>
